Filters/OrdinalDateFilter.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Fluid;
+using Fluid.Values;
+using OrchardCore.Liquid;
+using System;
+using System.Threading.Tasks;
+
+namespace Etch.OrchardCore.Liquid.Filters
+{
+    public class OrdinalDateFilter : ILiquidFilter
+    {
+        public ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, TemplateContext ctx)
+        {
+            if (input.IsNil())
+            {
+                return new ValueTask<FluidValue>(new StringValue("Invalid Date passed to ordinal filter"));
+            }
+
+            var date = (DateTime)input.ToObjectValue();
```

Comment:
I'd suggest moving the code that calculates the order to a separate private method, which is passed in to the return value. So it'll be...

```
return new ValueTask<FluidValue>(new StringValue(date.ToString($"MMMM d{CalculateOrdinal((DateTime)input.ToObjectValue())} yyyy")));
```